### PR TITLE
fixed setListeners not working on other locales

### PR DESCRIPTION
**Pull request by smaznet**
i had a issue which events didn't working
i put _setMethodHandler in try catch and i got date format exception
i noticed in DateTime to String you didn't localize date and in parseFrom String flutter tries parse date with device locale so error throws before firing events

### CODE REVIEW

File: pubspec.yaml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 name: awesome_notifications
 description: A complete solution to create Local Notifications and Push Notifications (Media Notifications, Big Picture, Big text, etc), through Firebase or another services, using Flutter.
-version: 0.7.0-beta.2
+version: 0.7.0-beta.4
```
Comment (webuijorgeGL) [inline pubspec.yaml:3]: the next beta version is 3 or I'm wrong?

**Comment by webuijorgeGL** [comment]
Looks like this issue happens on Android 10 and not later.

**Comment by smaznet** [comment]
> Looks like this issue happens on Android 10 and not later.

no i was testing on Android 12

File: pubspec.yaml (hunk 1 above)
Comment (smaznet) [reply pubspec.yaml:3]: because of my previous commit

**Comment by rokk4** [comment]
https://github.com/rafaelsetragni/awesome_notifications/issues/211 could this be possible related? Because it seems the issue only shows up in other timezones?

**Review by rafaelsetragni — APPROVED** [review]

**Comment by rafaelsetragni** [comment]
Finally done! Thank you so much and welcome aboard!

**Pull request merged**